src/main/java/edu/ucsb/mapache/google/Url.java

```diff
@@ -0,0 +1,52 @@
+package edu.ucsb.mapache.google;
+import java.util.Objects;
+
+public class Url{
+    private String type;
+    private String template;
+
+
+    public Url() {
+    }
+
+    public Url(String type, String template) {
+        this.type = type;
+        this.template = template;
+    }
+
+    public String getType() {
+        return this.type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getTemplate() {
+        return this.template;
+    }
+
+    public void setTemplate(String template) {
+        this.template = template;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Why is the input type for the `equals` method `Object`? (This comment goes for all of them, I'm just pointing out one instance). Shouldn't it be the same as the class instead(In this case the input type be `Url`)? It doesn't make sense to be comparing a `Url` to a `String` for example, so I don't think we should allow for it. It's a minor thing though.

Reply:
`Object#equals` is overridden from `Object`, so the method signature cannot be changed. `equals` must work even when you don't immediately know the types of the inputs (such as when they are stored in `Object` variables)